docs/source/upgrade.rst

```diff
@@ -0,0 +1,747 @@
+
+The ``chainer.config`` manages *the thread-local configuration*.
+You can also set the global configuration by modifying ``chainer.global_config``.
+Note that the global configuration is used only if the entry of the thread-local configuration is not explicitly set up.
+
+
+Variable
+--------
+
+.. _upgrade-volatile-removed:
+
+Volatile flag is removed
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :attr:`Variable.volatile` flag has been removed since Chainer v2.
+
+Instead, the configuration ``chainer.config.enable_backprop`` can be used to enable/disable the automatic differentiation feature.
+If it is ``True``, Chainer always creates a computational graph on the forward propagation, which corresponds to passing non-volatile variables in Chainer v1.
+Otherwise, Chainer does not create a graph, which corresponds to passing volatile variables in Chainer v1.
+The biggest difference is that ``enable_backprop`` is a thread-local flag, whereas ``volatile`` was a flag local to each :class:`Variable` object.
+Note that ``enable_backprop`` flag has already existed in Chainer v1, which took effect only if all the inputs to the function have ``volatile == 'auto'``.
+
+The ``chainer.config.enable_backprop`` flag can be modified directly or by using :func:`~chainer.using_config`.
+See :ref:`configuration` for details.
+There is also a convenience function, :func:`no_backprop_mode`, to turn off the flag.
+
+If you are using the ``Variable.volatile`` flag, you have to stop setting this flag (it will not take effect), and set the ``enable_backprop`` flag instead.
+
+.. admonition:: Example
+
+   Let ``model`` be your model, and consider the following code that calls it in volatile mode.
+
+   .. code-block:: py
+
+      # Chainer 1.x
+      x_data = ...   # ndarray
+      x = chainer.Variable(x_data, volatile=True)
+      y = model(x)
+
+   In Chainer 2.x, it should be updated as follows.
+
+   .. code-block:: py
+
+      # Chainer 2.x
+      x_data = ...   # ndarray
+      x = chainer.Variable(x)
+      with chainer.no_backprop_mode():
+          y = model(x)
+
+.. _upgrade-variable-node:
+
+Variable is not a part of a computational graph anymore
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :class:`Variable` class has been separated into two distinct classes, the ``Variable`` class and the :class:`VariableNode` class, since Chainer v2.
+Every class:`Variable` object owns its own :class:`VariableNode` object.
+A computational graph consists of :class:`Function` objects and :class:``VariableNode` objects.
+When one applies a :class:`Function` to a :class:`Variable`, the :class:`VariableNode` object of the variable is extracted and set to one of the inputs of the function.
+
+Note that the underlying data array of the variable is till held by the :class:`Variable` object.
+It allows each :class:`Function` implementation to release unneeded arrays from the computational graph, resulting in greatly reduced memory consumption.
```

Comment:
I meant "greatly reduced memory consumption" is one noun phrase. Maybe it's better to ask for native speakers.